### ISSUE TITLE
Stop squeezing stats by default

### ISSUE THREAD

**Issue by michaelosthege** [pymc-devs/pymc#6207]
Squeezing stats by default removes the `"stepmethod"` dimension (which we don't name explicitly), but this does not always work [for `object`-dtyped stats](https://github.com/pymc-devs/pymc/blob/d47dac0e92da2398206b42d3748bc63ff7ff7103/pymc/backends/base.py#L210-L215).

This is (very often) the case for the "warning" stat introduced by #6192, but also for stats that have varying shape.

We should probably not only stop squeezing, but instead unpack stats into a long form, such that there is not even a `stepmeethod` dimension.
This is how McBackend stores stats, because it works for any (hierarchical) combination of step methods.

**Comment by ricardoV94** [comment]
I think we have already done this

**Comment by michaelosthege** [comment]
No, it's still there: https://github.com/pymc-devs/pymc/blob/904a0eaac216732bc358dd91680cd428d95704f0/pymc/backends/base.py#L533

I was the last one to edit that file (7 months ago)

**Comment by ricardoV94** [comment]
I was thinking of this https://github.com/pymc-devs/pymc/commit/8ea8b3b6c2ceb67b964478783049888bb2ce44c1
